### DISCUSSION TITLE
x-model on select with pre-selected option does not select it

### DISCUSSION THREAD

**Discussion by jeannemas** [alpinejs/alpine#1872] [7. Bugs (archived)] [answered]
I've ran into an issue where using `x-model` on a `select` element where there is already a value, does not select the correct `option` (it selects the 1st `option` available in the list, no matter it's value). It seems to only appear while dynamically generating options using `x-for`.

An example in the codepen below.

I'm using the latest AlpineJS version available (`3.2.2`)

https://codepen.io/samariepled/pen/XWREpYd

**Comment by SimoTod** [comment, marked as answer]
you need to add `:selected="option == selected"` to your option tags. When x-model runs the first time, the select is empty (x-for runs after) so it can't select the right option. It was the same in V2

**Reply by jeannemas** [reply]
Thanks, this has been bugging me out for the last few days.
And since it was not documented anywhere, I really tought it was a bug.

**Comment by SergeyTsalkov** [comment]
If this can't be fixed, I think the documentation at https://alpinejs.dev/directives/model#dynamically-populated-select-options should mention this quirk. Currently the example in the docs is misleading and won't work as expected.

**Reply by ekwoka** [reply]
This is quite old. I'm also not sure what you mean it won't work?

**Reply by SergeyTsalkov** [reply]
Here's a test of the example at https://alpinejs.dev/directives/model#dynamically-populated-select-options:
https://codepen.io/Meekro/pen/BaqMeWM

When the page loads, the dropdown says Red and the text says Orange -- it's the same bug discussed in this thread, which is still a problem.

Now here's the same example fixed with the workaround from this thread (adding :selected to the option tag): https://codepen.io/Meekro/pen/wvYNbqZ

I'm suggesting that if this bug can't be fixed so the original example works, then the documentation should be modified. Maybe my corrected example should go in the documentation instead, or there should be a note or something?
